### ISSUE TITLE
budget forcing

### ISSUE THREAD

**Issue by Kenwwww** [simplescaling/s1#117]
anyone knows how to limit thinking length in vllm openai server？ it seems like need to adit the logits processor.
